mmv1/products/metastore/api.yaml

```diff
@@ -0,0 +1,235 @@
+      - !ruby/object:Api::Type::Enum
+        name: 'tier'
+        description: |
+          The tier of the service.
+        default_value: :TIER_UNSPECIFIED
+        values:
+          - :TIER_UNSPECIFIED
+          - :DEVELOPER
+          - :ENTERPRISE
+      - !ruby/object:Api::Type::NestedObject
+        name: 'maintenanceWindow'
+        description: |
+          The one hour maintenance window of the metastore service.
+          This specifies when the service can be restarted for maintenance purposes in UTC time.
+        properties:
+          - !ruby/object:Api::Type::Integer
+            name: 'dayOfWeek'
+            description: |
+              The hour of day (0-23) when the window starts.
+          - !ruby/object:Api::Type::Enum
+              name: 'optionalComponents'
+              description: |
+                The day of week, when the window starts.
+              default_value: :DAY_OF_WEEK_UNSPECIFIED
+              values:
+              - :DAY_OF_WEEK_UNSPECIFIED
+              - :MONDAY
+              - :TUESDAY
+              - :WEDNESDAY
+              - :THURSDAY
+              - :FRIDAY
+              - :SATURDAY
+              - :SUNDAY
+      - !ruby/object:Api::Type::String
+        name: 'uid'
+        output: true
+        description: |
+          The globally unique resource identifier of the metastore service.
+      - !ruby/object:Api::Type::NestedObject
+        name: 'metadataManagementActivity'
+        description: |
+         The metadata management activities of the metastore service.
+        properties:
+          - !ruby/object:Api::Type::Array
+            name: metadataExports
+            output: true
+            description: |
+              The latest metadata exports of the metastore service.
+            item_type: !ruby/object:Api::Type::NestedObject
+              properties:
+                - !ruby/object:Api::Type::Time
+                  name: 'startTime'
+                  description: |
+                    The time when the export started..
+                  output: true
+                - !ruby/object:Api::Type::Time
+                    name: 'endTime'
+                    description: |
+                      The time when the export ended.
+                    output: true
+                - !ruby/object:Api::Type::String
+                  name: 'state'
+                  description: |
+                    The current state of the export.
+                  output: true
+                - !ruby/object:Api::Type::String
+                  name: 'destinationGcsUri'
+                  description: |
+                     A Cloud Storage URI of a folder that metadata are exported to, in the form of `gs://<bucket_name>/<path_inside_bucket>/<export_folder>`.
+                  output: true
+      - !ruby/object:Api::Type::NestedObject
+        name: 'hiveMetastoreConfig'
```

Comment:
We should add either `required` or `at_least_one_of` to fields within nested objects to prevent users from setting empty blocks in their configs. Generally empty blocks are a headache and don't work correctly, so we try to either mark at least one field required or use `at_least_one_of` or `exactly_one_of`

Reply:
adding required to version !